### ISSUE TITLE
Running pre-trained with the other models like alexnet, googlenet

### ISSUE THREAD

**Issue by Crefeda** [leonardvandriel/caffe2_cpp_tutorial#69]
Hi,

I have used your pre-trained.cc on caffe2 model zoo on squeezenet models downloaded from caffe2's model zoo with the files: init.pb and predict_net.pb files
and I am able to successfully run it and get predictions. However, I have tried other models like googlenet and I face an error below and I am not sure why it is.

## Caffe2 Loading Pre-Trained Models Tutorial ##
https://caffe2.ai/docs/zoo.html
https://caffe2.ai/docs/tutorial-loading-pre-trained-models.html
https://caffe2.ai/docs/tutorial-image-pre-processing.html

init-net: convertmodels/init_net.pb
predict-net: convertmodels/predict_net.pb
file: res/image_file.jpg
size: 227

image size: [1280 x 751]
scaled size: [386 x 227]
cropped size: [227 x 227]
value range: (-128, 127)
*** Aborted at 1527159377 (unix time) try "date -d @1527159377" if you are using GNU date ***
PC: @           0x407590 caffe2::run()
*** SIGSEGV (@0x0) received by PID 12900 (TID 0x7feed7179b40) from PID 0; stack trace: ***
    @     0x7feed47244b0 (unknown)
    @           0x407590 caffe2::run()
    @           0x40526e main
    @     0x7feed470f830 __libc_start_main
    @           0x4052a9 _start
    @                0x0 (unknown)
Segmentation fault (core dumped)

And it seg faults at the following line: auto output = workspace.GetBlob("softmaxout")->Get<TensorCPU>();

I used the python script for downloading the models from the caffe2 model zoo and I also used your script to download the models from your repo.

Did you face similar issues?

Thanks,
C

**Comment by Crefeda** [comment]
Hi, 

It was the softmax layer that was causing the issue.
I changed it to this line:  auto output = workspace.GetBlob("prob")->Get<TensorCPU>();
I think there was a naming issue for the layer.

**Comment by leonardvandriel** [comment]
Thanks for figuring that out. I added a fix to the repo.
